pkg/kubecost/allocation.go

```diff
@@ -1194,140 +1177,120 @@ func (a *Allocation) generateKey(properties Properties) string {
 	// identifies allocations.
 	names := []string{}
 
-	if properties.HasCluster() {
-		cluster, _ := a.Properties.GetCluster()
-		names = append(names, cluster)
-	}
-
-	if properties.HasNode() {
-		node, _ := a.Properties.GetNode()
-		names = append(names, node)
-	}
-
-	if properties.HasNamespace() {
-		namespace, _ := a.Properties.GetNamespace()
-		names = append(names, namespace)
-	}
-
-	if properties.HasControllerKind() {
-		controllerKind, err := a.Properties.GetControllerKind()
-		if err != nil {
-			// Indicate that allocation has no controller
-			controllerKind = UnallocatedSuffix
-		}
-
-		if prop, _ := properties.GetControllerKind(); prop != "" && prop != controllerKind {
-			// The allocation does not have the specified controller kind
-			controllerKind = UnallocatedSuffix
-		}
-		names = append(names, controllerKind)
-	}
-
-	if properties.HasController() {
-		if !properties.HasControllerKind() {
-			controllerKind, err := a.Properties.GetControllerKind()
-			if err == nil {
-				names = append(names, controllerKind)
+	// Search for special case label for ETL conversion
+	aggControllerKind := ""
+	for _, agg := range aggregateBy {
+		if strings.HasPrefix(agg, "controllerKind:") {
+			aggControllerKind = strings.Split(agg, ":")[1]
+		}
+	}
```

Comment:
I have a sense, but can you document how this code block is designed to work, precisely? (e.g. what are the legal values and what happens when legal and illegal values are passed?) And is there a reason for doing it this way, as opposed to simply letting users pass `daemonset`, `deployment`, or `statefulset` as opposed to `controllerKind:...`?